### ISSUE TITLE
Fine Tuning

### ISSUE THREAD

**Issue by tshepang-p** [Mustard-Core/Career-Route-AI#32]

**Comment by tshepang-p** [comment]
@Kgotlello943 and @Mzwakhe-H Please work together and use hyper parameter tuning to try to improve our model accuracy, The model build is under the model repo, use the linear regression file

PLEASE DO NOT EDIT THE CODE, you can copy and paste onto your own file

**Comment by Kgotlello943** [comment]
Model was upgraded from Linear Regression to Ridge Regression because ridge Regression is a tuneable version of Linear Regression that uses L2 regularization to prevent overfitting, resulting in a much more stable and accurate salary prediction (R2 score improved to 0.87)

**Comment by Kgotlello943** [comment]
[tuned_ridge_reg.py](https://github.com/user-attachments/files/22958192/tuned_ridge_reg.py)

Implemented GridSearchCV to tune the regularization parameter (`alpha`).
Fixed data preprocessing by using One-Hot Encoding and MinMaxScaler for better feature handling.

**Comment by tshepang-p** [comment]
@Kgotlello943 Don't share code here, commit to model branch

**Comment by tshepang-p** [comment]
@Kgotlello943 please share your results, this code does not run on my side
I get an error on line 42

**Comment by Kgotlello943** [comment]
@tshepang-p 
Best Hyperparameter Found = 0.1
R-squared  Score: 0.8729
Mean Absolute Error : 13,311.04

**Comment by Kgotlello943** [comment]
What is the error message?
